File: fund/Binary Search/bs.py
def binary_search(target, arr):
    start = 0
    end = len(arr) - 1

    while(start < end):
        mid = (start + end) // 2

        if target == arr[mid]:
            return mid
        elif target < arr[mid]:
            end = mid
        elif target > arr[mid]:
            start = mid + 1

    return start if arr[start] == target else -1

if __name__ == '__main__':
    arr = [5,8,12,18,25,27,33,37,52,74]
    index = binary_search(74, arr)

    print(f'the index of the targe element is {index}')

File: fund/Binary Search/bs.go
package main

import "fmt"

func binary_search(target int, arr []int) int {
	var start int = 0
	var end int = len(arr) - 1
	var mid int

	for start < end {
		mid = (start + end) / 2
		if arr[mid] == target {
			return mid
		} else if target < arr[mid] {
			end = mid
		} else if target > arr[mid] {
			start = mid + 1
		}
	}

	if arr[start] == target {
		return start
	} else {
		return -1
	}
}

func main() {
	arr := []int{1, 2, 5, 6, 7, 9, 12, 15, 18, 20}

	index := binary_search(18, arr)

	fmt.Println("index of the target element is", index)
}